pkg/providers/vsphere/workers.go

```diff
@@ -71,16 +70,3 @@ func machineTemplateMapping() yamlutil.Mapping[*vspherev1.VSphereMachineTemplate
 		},
 	)
 }
-
-func getMachineTemplate(ctx context.Context, client kubernetes.Client, name, namespace string) (*vspherev1.VSphereMachineTemplate, error) {
```

Comment:
nit: Why remove them from here just to copy them to the other file?
If you don't want them to be in this file because they are shared, move them to a new third file, maybe? Not sure what we gain by just having them in the controlplane file instead of here.

Reply:
It just makes more sense to me to have them defined in the CP file vs. the workers file, but from a development perspective, yeah it doesn't change anything. I added it to a new, separate file.